### PR TITLE
Support `max_duration` argument in `cupyx.time.repeat`

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

File: cupyx/time.py [hunk 1]
```diff
@@ -42,7 +45,9 @@ def __str__(self):
         return self.to_str(show_gpu=True)
 
 
-def repeat(func, args=(), kwargs={}, n=10000, *, name=None, n_warmup=10):
+def repeat(
+        func, args=(), kwargs={}, n_repeat=10000, *,
+        name=None, n_warmup=10, max_dulation=math.inf):
```
Comment (emcastillo) [inline cupyx/time.py:50]: typo here, should be `max_duration`

File: cupyx/time.py [hunk 2]
```diff
@@ -81,7 +88,12 @@ def repeat(func, args=(), kwargs={}, n=10000, *, name=None, n_warmup=10):
         ev2.synchronize()
         cpu_time = t2 - t1
         gpu_time = cupy.cuda.get_elapsed_time(ev1, ev2) * 1e-3
-        ts[0, i] = cpu_time
-        ts[1, i] = gpu_time
+        cpu_times.append(cpu_time)
+        gpu_times.append(gpu_time)
 
+        dulation += cpu_time
+        if dulation > max_dulation:
```
Comment (emcastillo) [inline cupyx/time.py:95]: `duration`🙂

**Comment by emcastillo** [comment]
The PR looks good, lets fix the arg name!

File: cupyx/time.py (hunk 1 above)
Comment (asi1024) [reply cupyx/time.py]: Fixed 😓

File: cupyx/time.py (hunk 2 above)
Comment (asi1024) [reply cupyx/time.py]: Fixed.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**